tests/test_simple_pipeline_executor.py

```diff
@@ -214,20 +214,27 @@ def test_from_pipeline_intermediates_differ(self):
         )
 
         with self.assertLogs("lsst", level="INFO") as cm:
-            quanta = executor.run(register_dataset_types=True)
+            quanta = executor.run(register_dataset_types=True, save_versions=False)
         # A dict is given to task a without change.
         # A returns a dict because it has not been told to do anything else.
         # That does not match the storage class so it will be converted
         # on put.
-        # b is given a TaskMetadata.
+        # b is given a dict, because that's what its connection asks for.
         # b returns a TaskMetadata because that's how we configured it, but
         # the butler expects a dict so it is converted on put.
-        self._test_logs(
-            cm.output, "dict", "dict", "lsst.pipe.base.TaskMetadata", "lsst.pipe.base.TaskMetadata"
-        )
+        self._test_logs(cm.output, "dict", "dict", "dict", "lsst.pipe.base.TaskMetadata")
 
         self.assertEqual(len(quanta), 2)
         self.assertEqual(self.butler.get("intermediate").to_dict(), {"zero": 0, "one": 1})
+
+        # TODO: this test fails because we've set things up to need to do a
+        # dict -> TaskMetadata conversion on write and a TaskMetadata -> dict
+        # conversion on read, but the JSON formatter natively returns a dict
+        # and it looks like the formatter logic sees that as the same as the
+        # desired type and thinks it's done.  The problem is that the
+        # TaskMetadata dict representation has some entries that the dict it
+        # converts to does not.
+        self.skipTest("Upstream bug in daf_butler or TaskMetadata.")
```

Comment:
It might be better to mark this as expectedFail rather than skipping it. At least then when I fix it in daf_butler the test will fail because it passes.

Reply:
Looks like unittest can only do expected failures of an entire `TestCase`, so I've dropped the skip and changed from `assertEqual` to `assertNotEqual`.  That will have the same effect of forcing a fix to this test when the behavior is fixed, even though it won't appear as nicely in the categorization of the tests.